### PR TITLE
Replace spaces with underscores for preset/tag names

### PR DESCRIPTION
**Pull request by aNOOBisTheGod**
Now, when user adds new preset or tag, he can enter name with spacebars and they will be replaced to underlining. And also they will store with underlining in database so weird things shouldn't happen. I think it's kinda useful. Tested. Works fine

### CODE REVIEW

File: frontend/src/ts/ape/endpoints/users.ts [hunk 1]
```diff
@@ -70,6 +70,7 @@ export default function getUsersEndpoints(
   }
 
   async function createTag(tagName: string): Ape.EndpointData {
+    tagName = tagName.replaceAll(" ", "_");
```
Comment (Bruception) [inline frontend/src/ts/ape/endpoints/users.ts:73]: Avoid reassignment of arguments

File: frontend/src/ts/constants/firebase-config-example.ts [hunk 2]
```diff
@@ -4,11 +4,11 @@
...
 export const firebaseConfig = {
-  apiKey: "",
-  authDomain: "",
-  databaseURL: "",
-  projectId: "",
-  storageBucket: "",
-  messagingSenderId: "",
-  appId: "",
-};
+    apiKey: "",
```
Comment (Bruception) [inline frontend/src/ts/constants/firebase-config-example.ts:7]: Whats up with this file change?

File: frontend/src/ts/popups/edit-preset-popup.ts [hunk 3]
```diff
@@ -90,22 +90,24 @@ async function apply(): Promise<void> {
   Loader.show();
 
   if (action === "add") {
-    const response = await Ape.presets.add(presetName, configChanges);
-
+    const response = await Ape.presets.add(
+      presetName.replaceAll(" ", "_"),
```
Comment (Bruception) [inline frontend/src/ts/popups/edit-preset-popup.ts:94]: Why are we doing this if it already happens internally?

File: frontend/src/ts/popups/edit-preset-popup.ts [hunk 4]
```diff
@@ -90,22 +90,24 @@ async function apply(): Promise<void> {
...
+      configChanges
+    );
     if (response.status !== 200) {
       Notifications.add("Failed to add preset: " + response.message, -1);
     } else {
       Notifications.add("Preset added", 1, 2);
       snapshotPresets.push({
-        name: presetName,
+        name: presetName.replaceAll(" ", "_"),
```
Comment (Bruception) [inline frontend/src/ts/popups/edit-preset-popup.ts:102]: Try to keep this code DRY

File: frontend/src/ts/popups/edit-tags-popup.ts [hunk 5]
```diff
@@ -73,7 +73,7 @@ async function apply(): Promise<void> {
   Loader.show();
 
   if (action === "add") {
-    const response = await Ape.users.createTag(tagName);
+    const response = await Ape.users.createTag(tagName.replaceAll(" ", "_"));
```
Comment (Bruception) [inline frontend/src/ts/popups/edit-tags-popup.ts:76]: Why? If this happens internally

File: frontend/src/ts/ape/endpoints/users.ts (hunk 1 above)
Comment (aNOOBisTheGod) [reply frontend/src/ts/ape/endpoints/users.ts:73]: should I remake this?

File: frontend/src/ts/popups/edit-preset-popup.ts (hunk 3 above)
Comment (aNOOBisTheGod) [reply frontend/src/ts/popups/edit-preset-popup.ts:94]: > Why are we doing this if it already happens internally?

to be confident that there will be no spacebars in the database

Comment (Bruception) [reply frontend/src/ts/popups/edit-preset-popup.ts:94]: Calling the same function twice doesn't increase confidence.

Comment (Bruception) [reply frontend/src/ts/popups/edit-preset-popup.ts:94]: Calling the same function twice doesn't increase confidence.

File: frontend/src/ts/ape/endpoints/users.ts (hunk 1 above)
Comment (Bruception) [reply frontend/src/ts/ape/endpoints/users.ts:73]: Just store in other variable:
`const normalizedTagName = tagName.replaceAll(" ", "_");`

**Pull request closed without merging**